### PR TITLE
[APR-190] Add zstd compression level option

### PR DESCRIPTION
**Pull request by StephenWakely**
### What does this PR do?

Adds a new config option to set the zstd compression level, defaulting to the default level (5). Previously the default level was always used.

### Motivation

Using the default level has the potential to use up a lot more memory. Initial experiments show that using a lower level - ie 1, uses a lot less and still produces a significant improvement in compression ratio over zlib.

### Possible Drawbacks / Trade-offs

Given that this is a configurable option, no difference should be noticed in behaviour unless the user chooses to do so.

### Describe how to test/QA your changes

Add the following config options to `datadog.yml`:

```yml
serializer_compressor_kind: zstd
serializer_zstd_compressor_level: 1
```

### CODE REVIEW

File: pkg/config/setup/config.go [hunk 1]
```diff
@@ -1135,6 +1136,7 @@ func serializer(config pkgconfigmodel.Config) {
 	config.BindEnvAndSetDefault("serializer_max_series_payload_size", 512000)
 	config.BindEnvAndSetDefault("serializer_max_series_uncompressed_payload_size", 5242880)
 	config.BindEnvAndSetDefault("serializer_compressor_kind", DefaultCompressorKind)
+	config.BindEnvAndSetDefault("serializer_compressor_level", zstd.DefaultCompression)
```
Comment (remeh) [inline pkg/config/setup/config.go:1139]: I think we should name it differently since it applies to zstd only, WDTY? `serializer_zstd_compressor_level`?

**Comment by remeh** [comment]
On a separate note, I think all these `go.mod` updates comes with the fact that you include `DataDog/zstd` in the config package, which I think we don't want: some builds don't want zstd around (CGo).
A trade-off would be to declare a const `DefaultZstdCompressionLevel = 5` directly in `config.go`, WDYT?

**Review by remeh — APPROVED** [review]

**Review by rayz — APPROVED** [review]
lgtm for AML files

**Review by pgimalac — APPROVED** [review]

**Pull request merged**